dvc/tree/osf.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+import threading
+
+from funcy import cached_property, wrap_prop
+
+from dvc.exceptions import DvcException
+from dvc.hash_info import HashInfo
+from dvc.path_info import URLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+class OSFAuthError(DvcException):
+    def __init__(self):
+        message = (
+            "OSF authorization failed. Please check or provide"
+            " osf_username and password."
+            " It is also possible that you do not have access"
+            " to a project."
+        )
+        super().__init__(message)
+
+
+class OSFTree(BaseTree):
+    scheme = Schemes.OSF
+    REQUIRES = {"ofsclient": "osfclient"}
```

Comment:
Hi. So this will require the user to `pip install osfclient` themselves? Or can it be an optional dependency when you `pip install dvc[osf]`?